tests/test_cur.py

```diff
@@ -0,0 +1,86 @@
+import h5py
+import numpy as np
+
+from numpy.testing import assert_equal
+from numpy.testing import assert_array_equal
+from pyuoi.decomposition import CUR
+
+X = np.array([
+    [0, 0, 0, 4, 2],
+    [0, 4, 2, 1, 0],
+    [3, 0, 0, 2, 1],
+    [2, 2, 0, 1, 0],
+    [1, 2, 4, 1, 3],
+    [1, 4, 0, 0, 4],
+    [3, 3, 4, 0, 0],
+    [3, 2, 3, 0, 4],
+    [0, 1, 2, 1, 4],
+    [1, 4, 0, 2, 4]])
+
+
+def test_column_select_all():
+    """Test that column select function selects all columns when provided the
+    entire SVD and a suitable value of c."""
+    X = np.random.randint(low=1, high=5, size=(10, 5))
+    _, _, V = np.linalg.svd(X)
+    column_flags = CUR.column_select(V.T, c=5)
+
+    assert_array_equal(column_flags, np.array([True, True, True, True, True]))
+
+
+def test_column_select():
+    """Test that the column select function selects the vector with the highest
+    leverage score most often."""
+    n_samples, n_features = X.shape
+    rank = 3
+    n_reps = 5000
+
+    _, _, V = np.linalg.svd(X)
+    V_subset = V[:rank].T
+    column_flags = np.zeros((n_reps, n_features))
+
+    for rep in range(n_reps):
+        column_flags[rep] = CUR.column_select(V_subset, c=1)
+
+    counts = np.sum(column_flags, axis=0)
+
+    assert_equal(np.argmax(counts), np.argmax(np.sum(V_subset**2, axis=1)))
+
+
+def test_UoI_CUR_basic():
+    """Test UoI CUR with no resampling."""
+    n_samples, n_features = X.shape
+    max_k = 3
+    n_resamples = 1
+    resample_frac = 1
+
+    _, _, V = np.linalg.svd(X)
+    V_subset = V[:max_k].T
+
+    uoi_cur = CUR(n_resamples=n_resamples,
+                  max_k=max_k,
+                  resample_frac=resample_frac)
+    uoi_cur.fit(X, c=3)
+
+    max_col = np.argmax(np.sum(V_subset**2, axis=1))
+
+    assert (max_col in uoi_cur.columns_)
+
+
+def test_UoI_CUR_vs_CUR():
+    with h5py.File('tests/cur.h5', 'r') as data:
```

Comment:
Is it possible to just generate this h5 file on the fly? That's what I did here
https://github.com/BouchardLab/PyUoI/blob/8c422c64c8b5307dd7240e78b0b6126c38b5ae10/tests/test_mpi/test_mpi_utils.py#L15

Reply:
I'm not sure what you mean by this. I got this dataset from Shashanka, who used it for his testing. I'm not sure if there's anything special about it.